### PR TITLE
Add node rotation

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: internal/daemon/cluster/handlers/multihop_service.go [hunk 1]
```diff
@@ -80,3 +81,22 @@ func (m *multihopServiceServer) GenerateServerCertificates(ctx context.Context,
 		return multihopClient.GenerateServerCertificates(ctx, req)
 	}
 }
+
+func (m *multihopServiceServer) RotateNodeCredentials(ctx context.Context, req *types.RotateNodeCredentialsRequest) (*types.RotateNodeCredentialsResponse, error) {
```
Comment (irenarindos) [inline internal/daemon/cluster/handlers/multihop_service.go:85]: Could you add documentation to the functions in multihop_service?

**Review by jimlambrt — APPROVED** [review]
Great work!  (small minor suggestions)

File: internal/daemon/worker/auth_rotation.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	"github.com/hashicorp/boundary/internal/observability/event"
+	"github.com/hashicorp/nodeenrollment"
+	"github.com/hashicorp/nodeenrollment/multihop"
+	"github.com/hashicorp/nodeenrollment/types"
+)
+
+func (w *Worker) startAuthRotationTicking(cancelCtx context.Context) {
+	const op = "worker.(Worker).startAuthRotationTicking"
+	if w.conf.WorkerAuthKms != nil {
+		// Nothing to do here, just immediately exit
```
[the remark this answers is not in the document]
Comment (jefferai) [reply internal/daemon/worker/auth_rotation.go:18]: Probably, but I'm not sure the best way to do that -- via cancelCtx?

Generally speaking I'm also planning on going through all of the BYOW stuff once we have eventing sorted out for workers and adding stuff.

File: internal/daemon/worker/auth_rotation.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+			if earliestValid.After(now) || latestValid.Before(now) || earliestValid.After(latestValid) {
+				// We basically have no valid creds so we can't rotate.
+				// TODO (maybe): Have this trigger a new set of creds and request?
+				event.WriteSysEvent(cancelCtx, op, "not within node creds validity period, unsure what to do, not rotating")
+				continue
+			}
+
+			// Figure out the midpoint; if we're after it, try to rotate
+			delta := latestValid.Sub(earliestValid)
+			shouldRotate := now.Before(earliestValid.Add(delta / 2))
```
[the remark this answers is not in the document]
Comment (jefferai) [reply internal/daemon/worker/auth_rotation.go:91]: Done!

**Pull request merged**